src/libstd/sys/common/util.rs

```diff
@@ -46,11 +46,24 @@ unsafe fn abort_internal() -> ! {
     libc::abort()
 }
 
-// On Windows, we want to avoid using libc, and there isn't a direct
-// equivalent of libc::abort.  The __failfast intrinsic may be a reasonable
-// substitute, but desireability of using it over the abort instrinsic is
-// debateable; see https://github.com/rust-lang/rust/pull/31519 for details.
-#[cfg(not(unix))]
+// On windows, use the processor-specific __fastfail mechanism
+// https://msdn.microsoft.com/en-us/library/dn774154.aspx
```

Comment:
Could this elaborate on why we're using this? For example mention that exception handlers are removed and such, and also that on Windows 7 and prior we do *not* see this behavior but it largely equates to the same semantics.